docs/src/pages/components/data-grid/group-pivot/group-pivot.md

```diff
@@ -6,13 +6,113 @@ title: Data Grid - Group & Pivot
 
 <p class="description">Use grouping, pivoting and more to analyse the data in depth.</p>
 
-## 🚧 Tree data [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)
+## Tree Data [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)
 
-> ⚠️ This feature isn't implemented yet. It's coming.
->
-> 👍 Upvote [issue #210](https://github.com/mui-org/material-ui-x/issues/210) if you want to see it land faster.
+Tree Data allows to display data with parent / child relationships.
+
+### General behavior
+
+To enable the Tree Data, you simply have to use the `treeData` prop as well as provide a `getTreeDataPath` prop.
+The `getTreeDataPath` function returns an array of strings which represents the path to a given row.
+
+```tsx
+// The following examples will both render the same tree
+// - Sarah
+//     - Thomas
+//         - Robert
+//         - Karen
+
+const columns: GridColumns = [{ field: 'jobTitle', width: 250 }];
+
+// Without transformation
+const rows: GridRowsProp = [
+  { path: ['Sarah'], jobTitle: 'CEO', id: 0 },
+  { path: ['Sarah', 'Thomas'], jobTitle: 'Head of Sales', id: 1 },
+  { path: ['Sarah', 'Thomas', 'Robert'], jobTitle: 'Sales Person', id: 2 },
+  { path: ['Sarah', 'Thomas', 'Karen'], jobTitle: 'Sales Person', id: 3 },
+];
+
+<DataGridPro
+  treeData
+  getTreeDataPath={(row) => row.path}
+  rows={rows}
+  columns={columns}
+/>;
+
+// With transformation
+const rows: GridRowsProp = [
+  { path: 'Sarah', jobTitle: 'CEO', id: 0 },
+  { path: 'Sarah/Thomas', jobTitle: 'Head of Sales', id: 1 },
+  { path: 'Sarah/Thomas/Robert', jobTitle: 'Sales Person', id: 2 },
+  { path: 'Sarah/Thomas/Karen', jobTitle: 'Sales Person', id: 3 },
+];
+
+<DataGridPro
+  treeData
+  getTreeDataPath={(row) => row.path.split('/')}
+  rows={rows}
+  columns={columns}
+/>;
+```
+
+{{"demo": "pages/components/data-grid/group-pivot/BasicTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Custom grouping column
+
+Use the `groupingColDef` prop to customize the rendering of the grouping column.
+
+{{"demo": "pages/components/data-grid/group-pivot/CustomGroupingColumnTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Group expansion
+
+Use the `defaultGroupingExpansionDepth` prop to expand all the groups up to a given depth when loading the data.
+If you want to expand the whole tree, set `defaultGroupingExpansionDepth = -1`
+
+{{"demo": "pages/components/data-grid/group-pivot/DefaultGroupingExpansionDepthTreeData.js", "bg": "inline", "defaultCodeOpen": false}}
+
+Use the `unstable_setRowExpansion` method on `apiRef` to programmatically set the expansion of a row.
```

Comment:
Is not weird to tell users to use an unstable method? About the naming, it may cause confusion when #211 is implemented.

Reply:
Yes it's probably not great to prefix it and document it.
As it is a very basic method that will probably never change, I suppose we can make it public.
I had in mind we would use the same expansion boolean and method for boths as discussed on Monday. But as we won't, I can rename it`setRowChildrenExpansion`. In which case I would rename the `node.expanded` into `node.childrenExpanded`

Reply:
Done